File: mobile/app/src/main/java/com/hoangcn/quizbattle/battles/models/StartSoloMatchRequest.java
package com.hoangcn.quizbattle.battles.models;

import java.util.List;

public class StartSoloMatchRequest {
    private final String contentType;
    private final Integer topicId;
    private final List<Integer> fixedQuestionIds;

    public StartSoloMatchRequest(String contentType, Integer topicId, List<Integer> fixedQuestionIds) {
        this.contentType = contentType;
        this.topicId = topicId;
        this.fixedQuestionIds = fixedQuestionIds;
    }
}

File: server/Models/Matchs/Requests/StartSoloMatchRequest.cs
using System.Text.Json.Serialization;
using Models.Matchs.Enums;

namespace Models.Matchs.Requests
{
    public class StartSoloMatchRequest
    {
        [JsonConverter(typeof(JsonStringEnumConverter))]
        public MatchContentType ContentType { get; set; }
        public int? TopicId { get; set; }
        public List<int>? FixedQuestionIds { get; set; }
    }
}
